lib/em-http/http_connection.rb

```diff
@@ -36,6 +36,62 @@ def connection_completed
     def unbind(reason=nil)
       @parent.unbind(reason)
     end
+
+    # TLS verification support, original implementation by Mislav Marohnić
+    # https://github.com/lostisland/faraday/blob/63cf47c95b573539f047c729bd9ad67560bc83ff/lib/faraday/adapter/em_http_ssl_patch.rb
+    def ssl_verify_peer(cert_string)
+      cert = nil
+      begin
+        cert = OpenSSL::X509::Certificate.new(cert_string)
+      rescue OpenSSL::X509::CertificateError
+        return false
+      end
+
+      @last_seen_cert = cert
+
+      if certificate_store.verify(@last_seen_cert)
+        begin
+          certificate_store.add_cert(@last_seen_cert)
+        rescue OpenSSL::X509::StoreError => e
+          raise e unless e.message == 'cert already in hash table'
+        end
+        true
+      else
+        raise OpenSSL::SSL::SSLError.new(%(unable to verify the server certificate for "#{host}"))
+      end
+    end
+
+    def ssl_handshake_completed
+      unless verify_peer?
+        warn "[WARNING; em-http-request] TLS hostname validation is disabled (use 'tls: {verify_peer: true}'), see" +
```

Comment:
Could this be conditional?  If a caller explicitly doesn't want to verify_peer and they do

```
EventMachine::HttpRequest.new(url, opts.merge({tls: {verify_peer: false}})
```

It seems unnecessary kick this warning out...

Reply:
Yeah, that's reasonable. Willing to put together a quick patch? :)

Reply:
Sure.  How about https://github.com/igrigorik/em-http-request/pull/341